Add pw reset to salt login and make link always visible on bountysource

mitnuh opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bountysource/core?pullRequest=1083) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bountysource/core?pullRequest=1083) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bountysource/core?pullRequest=1083) it.</sub>

mitnuh commented:
The first commit added the reset link too salt but relied on the existing reset password form on bountysource. 

The latest commit adds a password reset form to salt and tweaks the reset password email so the reset link is to whichever website the request was on.

It works like bountysource reset form but looks a bit more like the salt login form.

wkonkel commented on app/controllers/api/v1/people_controller.rb:
```diff
@@ -247,7 +247,13 @@ def request_password_reset
 
     if (person = Person.find_by_email(params[:email]))
-      person.send_email(:reset_password)
+      if URI(request.referer).host == URI.parse(Api::Application.config.salt_url).host
```
I don't think `request.referer` is reliable... could we instead pass a `is_salt=true` boolean as a parameter?

mitnuh replied:
Will do! Was hoping to do this today but tomorrow is more likely.

mitnuh commented:
Changed to use is_salt parameter and added a test for the people_controller. 

If you have suggestions or advice for more/better tests they'd be appreciated! I considered adding a test to mailer_spec.rb for the extra parameter added to the email template but wasn't sure it would be a useful test...

mitnuh commented:
I need to git rebase because these commits are on my master branch and they're cluttering an unrelated PR. Might need to close this and make a new PR.
